### PR TITLE
AggLayer documentation in Innovation & design section

### PR DESCRIPTION
**Pull request by kmurphypolygon**
Working with @Jaime-N to produce some docs on the AggLayer

### CODE REVIEW

**Comment by kmurphypolygon** [comment]
@Jaime-N is doing a review so I removed @hsutaiyu

File: docs/cdk/agglayer/agglayer-go.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+## Overview
+
+The Agglayer Golang service is a web service designed to receive zero-knowledge proofs from various CDK chains, verify their soundness, and send them to L1 for final verification.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-go.md]:
```suggestion
The Agglayer Golang is a web service designed to receive zero-knowledge proofs (ZKP) from CDK chains, ensure they are valid, and send them to the L1 verifier contract for final verification.
```

File: docs/cdk/agglayer/agglayer-go.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+!!! warning
+    This service is now deprecating in favor of the more robust and efficient [Rust implementation](agglayer-rs.md).
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-go.md]:
```suggestion
    This service is being deprecated and replaced with a more robust and efficient [Rust implementation](agglayer-rs.md).
```

File: docs/cdk/agglayer/agglayer-go.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Architecture
+
+The AggLayer Golang architecture supports interactions with multiple CDK chains for ZKP verification. 
+
+It uses PostgreSQL database for storage and interacts with both L1 and L2 chains through configured RPC nodes.
+
+The diagram below shows the full start up, running, and shutdown sequence for the application and its components.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-go.md]:
```suggestion
The AggLayer Golang architecture supports interoperability among connected CDK chains by using ZKPs to ensure safety. It uses a PostgreSQL database for storage and interacts with both L1 and L2 chains through configured RPC nodes.

The diagram below shows the full start-up, running, and shutdown sequence for the AggLayer Golang application and its components.
```

File: docs/cdk/agglayer/agglayer-go.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+## Production set up
+
+1. Ensure only one instance of Agglayer is running at a time.
+2. Use a containerized setup, or OS level service manager/monitoring system, for automatic restarts in the case of failures.
+
+### Prerequisites
+
+#### Hardware
+
+- For each CDK chain it's necessary to configure it's corresponding RPC node, synced with the target CDK.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-go.md]:
```suggestion
- For each CDK chain it's necessary to configure its corresponding RPC node, synced with the target CDK. 
```

File: docs/cdk/agglayer/agglayer-rs.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
+## Overview
+
+AggLayer is a Rust-based web service designed to enhance interoperability among heterogeneous blockchain networks by securely handling zero-knowledge proofs.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-rs.md]:
```suggestion
The AggLayer is a Rust-based web service designed to ensure safety for cross-chain asset transfers among heterogeneous blockchain networks. This safety is provided using ZKPs, allowing end-users to interoperate with no additional trust assumptions.  
```

File: docs/cdk/agglayer/agglayer-rs.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+The service verifies the soundness of proofs from various CDK chains before forwarding them to L1 for verification.
+
+It replaces the previous [Golang implementation](agglayer-go.md).
+
+## Architecture
+
+### Components
+
+- Aggregator: Receives and verifies zk-proofs.
+- Verifier: Confirms the soundness of proofs before submitting them to L1.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-rs.md]:
```suggestion
- Unified bridge: A shared bridge contract for all AggLayer-connected chains, allowing for the transfer of native assets.
- Pessimistic proof: A novel ZKP that provides safety for the assets on the unified bridge by ensuring no chain can withdraw more tokens of any type than have been deposited into it. The pessimistic proof is validated the verifier contract on the L1. 
```

File: docs/cdk/agglayer/agglayer-go.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+2. Execute the following command:
+
+    ```bash
+    make run-docker
+    ```
+
+## Production set up
+
+1. Ensure only one instance of Agglayer is running at a time.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-go.md]:
```suggestion
1. Ensure only one instance of the Agglayer is running at a time.
```

File: docs/cdk/agglayer/agglayer-rs.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
+## Overview
+
+The AggLayer is a Rust-based web service designed to ensure safety for cross-chain asset transfers among heterogeneous blockchain networks. This safety is provided using ZKPs, allowing end-users to interoperate with no additional trust assumptions.  
+
+The service verifies the soundness of proofs from various CDK chains before forwarding them to L1 for verification.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-rs.md]:
```suggestion
The service verifies the validity of proofs from various CDK chains before forwarding them to L1 for verification.
```

File: docs/cdk/agglayer/agglayer-rs.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+It replaces the previous [Golang implementation](agglayer-go.md).
+
+## Architecture
+
+### Components
+
+- Unified bridge: A shared bridge contract for all AggLayer-connected chains, allowing for the transfer of native assets.
+- Pessimistic proof: A novel ZKP that provides safety for the assets on the unified bridge by ensuring no chain can withdraw more tokens of any type than have been deposited into it. The pessimistic proof is validated the verifier contract on the L1.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-rs.md]:
```suggestion
- Unified bridge: A shared bridge contract for all AggLayer-connected chains, allowing for the transfer of fungible tokens. 
- Pessimistic proof: A novel ZKP that provides safety for the tokens deposited to the unified bridge by ensuring no chain can withdraw more tokens of any type than have been deposited into it. Pessimistic proofs are validated by the verifier contract on the L1. 
```

File: docs/cdk/agglayer/agglayer-rs.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+## Architecture
+
+### Components
+
+- Unified bridge: A shared bridge contract for all AggLayer-connected chains, allowing for the transfer of native assets.
+- Pessimistic proof: A novel ZKP that provides safety for the assets on the unified bridge by ensuring no chain can withdraw more tokens of any type than have been deposited into it. The pessimistic proof is validated the verifier contract on the L1. 
+
+### Data flow
+
+- Input: Proofs from various CDK chains.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/agglayer-rs.md]:
```suggestion
- Input: Proofs of execution generated by CDK chains.
```

File: docs/cdk/agglayer/bridge-and-call.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
+## Current unified bridge functionality
+
+The unified bridge allows users/contracts to independently bridge an asset or bridge a message between AggLayer connected chains.
+
+More complex operations such as bridge an asset and swap it for another asset, or bridge an asset and deposit it into a vault, require implementing custom logic.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/bridge-and-call.md]:
```suggestion
The unified bridge allows users to trustlessly bridge a fungible token or message between AggLayer-connected chains.

More complex operations, in which some action is taken on the destination chain with a bridged token, such as a swap, require implementing custom logic. 
```

File: docs/cdk/agglayer/overview.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
+!!! info "Disclaimer"
+    - Much of the content in this section is discussing technology in development and not ready for release.
+    - Please check against the main documentation site for any live releases.
+    - Feel free to experiment with any code in public repos.
+
+## Welcome to AggLayer documentation
+
+The culmination of Polygon Lab's innovation and design, AggLayer is an evolving ecosystem of zk-based (zero-knowledge) interacting chains. It addresses the current industry focus on interoperability and shared liquidity.
+
+While most rollup environments follow a modular approach, where developers pick-and-choose components such as execution and data availability layers for example, the Agglayer vision is uniquely centered around zk-proof technology.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/overview.md]:
```suggestion
The AggLayer is an in-development interoperability protocol that allows for trustless, cross-chain token transfers and message-passing, as well as more complex operations. The safety of the AggLayer is provided by zero-knowledge proofs.
```

File: docs/cdk/agglayer/agglayer-go.md (hunk 7 above)
Comment (kmurphypolygon) [reply docs/cdk/agglayer/agglayer-go.md]: updated over a previous update

File: docs/cdk/agglayer/agglayer-rs.md (hunk 9 above)
Comment (kmurphypolygon) [reply docs/cdk/agglayer/agglayer-rs.md]: querying this one

File: docs/cdk/agglayer/unified-bridge.md [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+!!! important "AggLayer smart contracts"
+    - The current version of the unified bridge uses the contracts in the [PolygonzkEVM smart contract repo](https://github.com/0xPolygonHermez/zkevm-contracts).
+
+## Bridging mechanism
+
+The bridging mechanism enables token transfers and message-passing between Ethereum (L1) and CDK chains via smart contracts. Detailed in the [zkEVM bridging documentation](../../zkEVM/architecture/high-level/smart-contracts/bridging.md), core components include the bridge and exit root Solidity smart contracts.
+
+## Data structures 
+
+Each chain holds a single data structure which stores a record of all token withdrawals and messages that originated from that chain. This “exit tree” is an append-only Merkle trie, similar in structure to the Ethereum deposit trie” The latest state of each chain and the unified bridge is represented by the root of this Merkle tree, referred to as the “exit root.”
```
Comment (Jaime-N) [inline docs/cdk/agglayer/unified-bridge.md]:
```suggestion
Each chain holds a single data structure which stores a record of all token withdrawals and messages that originated from that chain. This “exit tree” is an append-only Merkle trie, similar in structure to the Ethereum deposit trie. The latest state of each chain and the unified bridge is represented by the root of this Merkle tree, referred to as the “exit root.”
```

File: docs/cdk/agglayer/unified-bridge.md [hunk 14]
```diff
@@ -0,0 +1,18 @@
+AggLayer's beginnings lie in the Polygon zkEVM unified bridge technology which was originally implemented for Polygon zkEVM but now also supports CDK chains.
+
+The unified bridge accelerates Ethereum interoperability by using advanced cryptographic techniques and Solidity smart contracts on L1 and L2. It facilitates secure cross-chain interactions and asset transfers, ensuring transaction throughput and integrity.
+
+An Ethereum scalability solution, the unified bridge facilitates seamless L1-L2 interaction and relies on robust and secure zero-knowledge proofs.
```
Comment (Jaime-N) [inline docs/cdk/agglayer/unified-bridge.md]:
```suggestion
## Overview

The unified bridge is a single bridge contract on Ethereum, providing a safe, common access point for the transfer of all native, never-wrapped tokens. Each chain has a local copy of the unified bridge root, enabling cross-chain interoperability that doesn’t require the security risks of third-party bridges

```

**Review by Jaime-N — CHANGES_REQUESTED** [review]
I submitted two changes to the Unified Bridge material. Once those are approved, it's good to merge!

**Review by Jaime-N — APPROVED** [review]

**Pull request merged**